CRM/Event/BAO/Participant.php

```diff
@@ -1586,10 +1586,7 @@ public static function sendTransitionParticipantMail(
           'subject' => $subject,
           'source_contact_id' => $contactId,
           'source_record_id' => $participantId,
-          'activity_type_id' => CRM_Core_OptionGroup::getValue('activity_type',
-            $activityType,
-            'name'
-          ),
+          'activity_type_id' => CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_type_id', $activityType),
```

Comment:
ok ($activityType is actually hard coded as 'Event Registration' which is valid)